starlette/routing.py

```diff
@@ -224,6 +224,8 @@ async def handle(self, scope: Scope, receive: Receive, send: Send) -> None:
                 response = PlainTextResponse("Method Not Allowed", status_code=405)
             await response(scope, receive, send)
         else:
+            if "session" in scope:
+                await scope["session"].load()
```

Comment:
Automatically loading the session for *all* routes seems to be like it could be wasteful when using slower backends, and not necessarily something you would want everywhere. It's a problem `flask-login` has as well, which can completely tank performance in some instances, especially with mountpoints for static files, and doubly-so for sub-applications that could have their own session middleware running.

Reply:
What do you think if we add the "autoload" argument to the session middleware? So a developer can get a little more control over it? 
/cc @tomchristie